apiserver/facades/agent/instancemutater/instancemutater.go

```diff
@@ -39,6 +39,9 @@ type MachineService interface {
 	// It returns a MachineNotFound if the machine does not exist.
 	GetMachineUUID(ctx context.Context, name coremachine.Name) (string, error)
 
+	// InstanceID returns the cloud specific instance id for this machine.
+	InstanceID(ctx context.Context, mUUID string) (string, error)
```

Comment:
I'll refactor all methods from the machine domain to receive a custom UUID struct, also to handle `instance.Id` instead of strings. This will arrive in a follow-up PR